src/lib.rs

```diff
@@ -162,88 +187,138 @@ impl Drop for Gdbm {
     }
 }
 
+/// With locking disabled (if gdbm_open was called with ‘GDBM_NOLOCK’), the user may want
+/// to perform their own file locking on the database file in order to prevent multiple
+/// writers operating on the same file simultaneously.
+impl AsRawFd for Gdbm {
+    fn as_raw_fd(&self) -> RawFd {
+        unsafe {
+            gdbm_fdesc(self.db_handle) as RawFd
+        }
+    }
+}
+
 impl Gdbm {
     /// Open a DBM with location.
-    /// mode (see http://www.manpagez.com/man/2/chmod,
-    /// and http://www.manpagez.com/man/2/open), which is used if the file is created).
-    pub fn new(path: &Path, block_size: u32, flags: Flags, mode: i32) -> Result<Gdbm, GdbmError> {
-        let path = CString::new(path.as_os_str().as_bytes())?;
+    ///
+    /// `mode` is the unix mode used when a database is created. See
+    /// [chmod](http://www.manpagez.com/man/2/chmod) and
+    /// [open](http://www.manpagez.com/man/2/open).
+    pub fn new(
+        path: impl AsRef<Path>,
+        block_size: u32,
+        flags: Open,
+        mode: u32
```

Comment:
I know it would be probably overkill but if you wanted you could use nix::sys::stat::Mode here to make it a little more ergonomic. It would require pulling in the nix crate for just one type though....
It would also make this function a little easier to write. Lines 213 -> 218 could be skipped since the type restricts the possibilities.

Reply:
I think pulling in the nix crate is overkill.. There is precedent: the standard library [also uses a plain u32](https://doc.rust-lang.org/stable/std/os/unix/fs/trait.PermissionsExt.html#tymethod.set_mode).

Reply:
Oh ok. Perfect :)